link/query_test.go

```diff
@@ -0,0 +1,81 @@
+package link
+
+import (
+	"testing"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/internal/testutils"
+)
+
+func TestQueryPrograms(t *testing.T) {
+	for name, fn := range map[string]func(*testing.T) (*ebpf.Program, QueryOptions){
+		"cgroup": queryCgroupFixtures,
+		"netns":  queryNetNSFixtures,
+	} {
+		t.Run(name, func(t *testing.T) {
+			prog, opts := fn(t)
+			ids, err := QueryPrograms(opts)
+			testutils.SkipIfNotSupported(t, err)
+			if err != nil {
+				t.Fatal("Can't query programs:", err)
+			}
+
+			progInfo, err := prog.Info()
+			if err != nil {
+				t.Fatal("Can't get program info:", err)
+			}
+
+			progId, _ := progInfo.ID()
+
+			for _, id := range ids {
+				if id == progId {
+					return
+				}
+			}
+			t.Fatal("Can't find program ID in query")
```

Comment:
Nit: printing the relevant results in test output is quite convenient, e.g.:

```suggestion
			t.Fatalf("Can't find program ID %d in query result: %v", progId, ids)
```